qiskit/compiler/transpile.py

```diff
@@ -301,18 +267,29 @@ def _parse_basis_gates(basis_gates, backend, circuits):
     return basis_gates
 
 
-def _parse_coupling_map(coupling_map, backend, num_circuits):
+def _parse_coupling_map(coupling_map, backend, circuits):
+    num_circuits = len(circuits)
     # try getting coupling_map from user, else backend
     if coupling_map is None:
         if getattr(backend, 'configuration', None):
             coupling_map = getattr(backend.configuration(), 'coupling_map', None)
-    # coupling_map could be None, or a list of lists, e.g. [[0, 1], [2, 1]]
-    if coupling_map is None or isinstance(coupling_map, CouplingMap):
-        coupling_map = [coupling_map] * num_circuits
-    elif isinstance(coupling_map, list) and all(isinstance(i, list) and len(i) == 2
-                                                for i in coupling_map):
-        coupling_map = [coupling_map] * num_circuits
-    coupling_map = [CouplingMap(cm) if isinstance(cm, list) else cm for cm in coupling_map]
+
+    coupling_map = [coupling_map] * num_circuits
+
+    for index, parsed_coupling_map in enumerate(coupling_map):
+        if isinstance(parsed_coupling_map, list):
```

Comment:
Could we maybe save a little work by doing this once from the input `coupling_map` before multiplying by `num_circuits`?

Reply:
They will all have the same instance of coupling map... I'm thinking over the consequence of that. But it might be something positive...

Reply:
this situation handles when the input `coupling_map` is a list of coupling maps defined as a list. Therefore, they might be different among them. In 948b3d44, I construct a coupling map once if it's coming from the backend `CouplingMap(backend.configuration().coupling_map)`